packages/core/lib/debug/printer.js

```diff
@@ -209,11 +210,43 @@ class DebugPrinter {
       this.config.logger.log(DebugUtils.formatStack(step.stack));
       this.config.logger.log("");
     }
-    this.config.logger.log(
-      DebugUtils.formatInstruction(traceIndex + 1, totalSteps, instruction)
-    );
-    this.config.logger.log(DebugUtils.formatPC(step.pc));
+
+    this.config.logger.log("Instructions:");
+    this.config.logger.log("");
+
+    // printout instructions
+    // add an ellipse if there exist additional instructions before
+    const previousInstructions = 3;
+    const upcomingInstructions = 3;
+    const currentIndex = instruction.index;
+
+    if (currentIndex - previousInstructions >= 0) {
+      this.config.logger.log("...");
+    }
+    // printout 3 previous instructions
+    for (let i = Math.max(currentIndex - previousInstructions, 0);
+      i < currentIndex;
+      i++) {
+      this.config.logger.log(DebugUtils.formatInstruction(instructions[i]));
+    }
+
+    // printout current instruction
+    this.config.logger.log(DebugUtils.formatCurrentInstruction(instruction));
+
+    // printout 3 upcoming instructions
+    for (let i = Math.min(currentIndex + 1, instructions.length);
+      i <= currentIndex + upcomingInstructions;
+      i++) {
+      this.config.logger.log(DebugUtils.formatInstruction(instructions[i]));
+    }
+
+    // add an ellipse if there exist additional instructions after
+    if (currentIndex + upcomingInstructions < instructions.length) {
+      this.config.logger.log("...");
+    }
+
     this.config.logger.log("");
+    this.config.logger.log("Step ", + traceIndex + "/" + totalSteps);
```

Comment:
Sorry, just noticed two problems here!

1. Bigger problem: You forgot to add 1 to the traceIndex before printing.  The traceIndex is 0-indexed, but we print it out 1-indexed.  You'll notice the old code adds 1 to it; the new code should do the same.
2. Smaller problem: You have a stray comma in here that's causing an extra space to appear in your output.

Btw, in new code we generally use template strings rather than manual string concatenation.  I won't insist on that though... there's probably a lot of manual string concatenation around here already so it's hardly a big deal. :P

Reply:
thanks for catching these.  Changes made.

Reply:
Yeah, for reference, you can do this:
```javascript
console.log(`This is a ${"template"} string`);
```

Reply:
For a more illustrative example, in this case one could do `Step ${traceIndex + 1}/${totalSteps}`.